guides/request-validation-csharp/example-1/example-1.4.x.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Configuration;
+using System.Web;
+using System.Net;
+using System.Web.Mvc;
+using Twilio;
+using Twilio.TwiML;
+
+namespace helloworldcsharp
+{
+	public class RequestValidatorFilterAttribute : ActionFilterAttribute
+	{
+
+		private static readonly Lazy<RequestValidator> TwilioRequestValidator
+			= new Lazy<RequestValidator>(() => new RequestValidator());
+
+		public override void OnActionExecuting(ActionExecutingContext filterContext)
+		{
+			var twilioAuthToken = ConfigurationManager.AppSettings["TwilioAuthToken"];
```

Comment:
Usually is not a good idea to hardcode values in any implementation. This this implementation is coupled to the idea of having a key in the _application settings_ named "TwilioAuthToken".

A more idiomatic approach is to use an overloaded constructor. When no value is provided then we can rely on a fallback value "TwilioAuthToken", otherwise we can use the provided key through the constructor.

```csharp
private readonly string _twilioAuthTokenKey;

public ValidateTwilioRequestAttribute(string twilioAuthTokenKey)
{
    _twilioAuthTokenKey = twilioAuthTokenKey;
}

public ValidateTwilioRequestAttribute()
{
    _twilioAuthTokenKey = "TwilioAuthToken";
}

public override void OnActionExecuting(ActionExecutingContext filterContext)
{
    var twilioAuthToken = ConfigurationManager.AppSettings[_twilioAuthTokenKey];
    // ...

```